src/NuGet.Clients/NuGet.VisualStudio/Extensibility/IVsProjectJsonToPackageReferenceMigrator.cs

```diff
@@ -14,12 +15,13 @@ namespace NuGet.VisualStudio
     public interface IVsProjectJsonToPackageReferenceMigrator
     {
         /// <summary>
-        /// Migrates a legacy Project.json based project to Package Reference based project.
+        /// Migrates a legacy Project.json based project to Package Reference based project. The result 
+        /// should be casted to type <see cref="IVsProjectJsonToPackageReferenceMigrateResult"/>
         /// The backup of the original project file and project.json file is creatd in the Backup folder
         /// in the root of the project directory.
         /// </summary>
         /// <param name="projectUniqueName">The full path to the project that needs to be migrated</param>
-        IVsProjectJsonToPackageReferenceMigrateResult MigrateProjectJsonToPackageReference(string projectUniqueName);
+        Task<object> MigrateProjectJsonToPackageReferenceAsync(string projectUniqueName);
```

Comment:
Please re-guid the embeddable interface. Any and every time you change an embeddable interface, please, please re-guid it to make sure that if there is some failure, you get an InvalidCastException that directs you straight at the problem instead of a random AV crash.